pkg/kubelet/kuberuntime/labels.go

```diff
@@ -89,7 +92,20 @@ func newPodLabels(pod *v1.Pod) map[string]string {
 
 // newPodAnnotations creates pod annotations from v1.Pod.
 func newPodAnnotations(pod *v1.Pod) map[string]string {
-	return pod.Annotations
+	annotations := map[string]string{}
+
+	// Get annotations from v1.Pod
+	for k, v := range pod.Annotations {
+		annotations[k] = v
+	}
+
+	if kubecontainer.HasWindowsHostProcessContainer(pod) {
```

Comment:
should this logic only execute in Windows build of kubelet?

Reply:
same qq - whether it should run under feature flag only?

Reply:
This seems like a reasonable request.  The generate `generatePodSandboxWindowsConfig` is behind a GOOS flag and there is a feature flag check there as well. 

looks like we might want it https://github.com/kubernetes/kubernetes/pull/99576/files#diff-49d1c9b046a0055e2c7cd93a85f90131819eb5e30eaaac1daf3ad66a289e5af3R125-R126 and https://github.com/kubernetes/kubernetes/pull/99576/files#diff-49d1c9b046a0055e2c7cd93a85f90131819eb5e30eaaac1daf3ad66a289e5af3R125-R126 as well

The pod its self would be rejected due to https://github.com/kubernetes/kubernetes/pull/99576/files#diff-2e2d495cb2f6938efb55dfcd758c182922b2f28271f9c237bf710c49cb8a12a7R245 so those annotations would never make it through to actual create request to the runtime.

What do you think?

Reply:
all linked examples are pointing to either files with `_windows.go` or called under GOOS. I understand this code will never result in setting the label. It also will be removed in beta. This comment is just to be on a safe side and only call this under GOOS or with the feature flag. This is what I typically pay attention when review PRs introducing the new feature flags.

Reply:
Will make those updates

Reply:
I put this on a separate commit for easier review: https://github.com/kubernetes/kubernetes/pull/99576/commits/9fc16f4ed58f9ed2453e5b81216a36292686fe25

If it looks good I can rebase it into the other commits.